### PR TITLE
Accessibility: Add missing tooltips to icon buttons

### LINKED ISSUE #3384: Use tooltips for all the icon-only controls
Most of the controls in the toolbars use tooltips:

screen shot 2017-11-08 at 11 48 55

Some of them (typically buttons that open drop down menus) don't:

screen shot 2017-11-08 at 11 49 05

screen shot 2017-11-08 at 11 49 16

For consistency, I'd suggest to always use tooltip on all the controls.

### PR DESCRIPTION
**Pull request by youknowriad**
closes #3384

This PR adds the missing tooltips to icon-only buttons.

 - The block switcher
 - Some media upload buttons

If you notice another missing tooltip, let me know.

### CODE REVIEW

**Comment by afercia** [comment]
@youknowriad thanks!

1
Missing tooltips noticed so far:
- Classic block: all (I know this is TinyMCE but the current editor has working tooltips on all its buttons, and they should have tooltips in Gutenberg as well)
- Video block: pencil icon
- Audio block: pencil icon
- Table block: edit table

2
More related to design/consistency /cc @jasmussen : some tooltips have title-cased text, some not. Regardless of language-specific preferences (as far as I know title-case is preferred in English, but it's not in many other languages), there should be some consistency. Since these are tooltips and not button labels, I'd go for no title-casing. Some example of the current tooltips:
```
Insert block
Change block type
Align left
Move Up
Move Down
Edit Gallery
Edit image
Remove Image
Grid View
List View
```

3
Slightly unrelated, I guess it should be addressed elsewhere: when I insert an image, by default it has no link. However, the link button says "Edit link". In the classic editor, the link button always says "Insert/edit link" to avoid this issue.

4
Unrelated but worth mentioning: the image edit link "modal" has some issues...

screen shot 2018-01-05 at 14 36 11

**Comment by youknowriad** [comment]
I fixed the point "1". I tried to make the freeform block's tooltips look similarily to the default tooltips.

**Comment by youknowriad** [comment]
Other points are unrelated and may be resolved separately.

**Comment by afercia** [comment]
> Other points are unrelated and may be resolved separately.

Sure, will split them in a separate issue so this can be merged/closed.

File: blocks/library/gallery/block.js [hunk 1]
```diff
@@ -149,13 +150,14 @@ class GalleryBlock extends Component {
 							<MediaUploadButton
 								buttonProps={ {
 									className: 'components-icon-button components-toolbar__control',
-									'aria-label': __( 'Edit Gallery' ),
+									'aria-label': editButtonLabel,
```
Comment (aduth) [inline blocks/library/gallery/block.js]: Would be nice if we could infer this from the `tooltip` passed to `MediaUploadButton`, but I guess the upload button is not always an icon button, so not so simple.

File: components/icon-button/index.js [hunk 2]
```diff
@@ -23,6 +23,8 @@ class IconButton extends Component {
 	render() {
 		const { icon, children, label, className, tooltip, focus, ...additionalProps } = this.props;
 		const classes = classnames( 'components-icon-button', className );
+		const tooltipText = tooltip || label;
+		const showTooltip = !! tooltip || ( label && ! children && false !== tooltip );
```
Comment (aduth) [inline components/icon-button/index.js]: Edge case: Should `showTooltip` be `true` or `false` if `children` is `[]` ?

(Related: https://reactjs.org/docs/react-api.html#reactchildrencount )

**Review by aduth — APPROVED** [review]

File: blocks/library/gallery/block.js (hunk 1 above)
Comment (youknowriad) [reply blocks/library/gallery/block.js]: Actually I was planing to revisit the `MediaUploadButton` to use render Props instead, that way we'd just use an `IconButton` in the render prop. 

The thinking is that the `MediaUpload` should only deal with `MediaUpload` and not the UI to trigger the modal.

File: components/icon-button/index.js (hunk 2 above)
Comment (youknowriad) [reply components/icon-button/index.js]: mmm, I guess it's better if it's true.

**Comment by afercia** [comment]
I guess the following Trac ticket is relevant for the TinyMCE's tooltip: while the visual tooltips are translatable, the buttons aria-label attributes are not (not all, at least), see https://core.trac.wordpress.org/ticket/35710

Should it be tracked in a new issue?

**Comment by jasmussen** [comment]
> Should it be tracked in a new issue?

Is this a TinyMCE issue? CC: @ephox-mogran

**Pull request merged**